Create Blog Post “10-hot-summer-apps-you-can-build-with-open-source-apis” (#2549)
Nexmo/deved-platform: content/blog/en/10-hot-summer-apps-you-can-build-with-open-source-apis.md

---
title: 10 Hot Summer Apps You Can Build with Open-Source APIs
description: Uncover 10 exciting summer-themed app ideas that leverage
  open-source APIs and the joy of coding with your favorite summer activities.
author: diana-pham-1
published: true
published_at: 2023-08-01T20:14:35.232Z
updated_at: 2023-08-01T20:14:35.245Z
category: inspiration
tags:
  - inspiration
comments: true
spotlight: false
redirect: ""
canonical: ""
outdated: false
replacement_url: ""
---
As a beacon of code and a big fan of summer, I've been exploring ways to mix enjoying the season with a bit of coding. The sun is out, and here are some hot apps you can build to pair with your favorite summer activities.

## Have Some Ice Cream with Your IDE

> 1. **Hydration Manager**: Thirst isn't always a reliable early indicator of the body's need for water. Many people don't feel thirsty until they're already dehydrated. To make sure you stay hydrated this summer, you can [build an app using the Vonage Messages API]((https://developer.vonage.com/en/blog/drinking-water-reminder-app-with-node-react-and-vonage-messages-api) ) that sends friendly reminders to drink up!

Reviewer: ```suggestion
1. **Hydration Manager**: Thirst isn't always a reliable early indicator of the body's need for water. Many people don't feel thirsty until they're already dehydrated. To make sure you stay hydrated this summer, you can [build an app using the Vonage Messages API](https://developer.vonage.com/en/blog/drinking-water-reminder-app-with-node-react-and-vonage-messages-api) that sends friendly reminders to drink up!
```

> 1. **Hydration Manager**: Thirst isn't always a reliable early indicator of the body's need for water. Many people don't feel thirsty until they're already dehydrated. To make sure you stay hydrated this summer, you can build an app using the Vonage Messages API that sends friendly reminders to drink up! Here's a [blog](https://developer.vonage.com/en/blog/drinking-water-reminder-app-with-node-react-and-vonage-messages-api) on how to create that app.

Reviewer: As per our guide https://learn.vonage.com/contributing/: For links, try to have the link naturally included in an appropriate phrase. Rather than using “You can find more information here” or “Click here to read more”, use something like “The SMS API allows you to quickly send messages.”

> 1. **Hydration Manager**: Thirst isn't always a reliable early indicator of the body's need for water. Many people don't feel thirsty until they're already dehydrated. To make sure you stay hydrated this summer, you can build an app using the Vonage Messages API that sends friendly reminders to drink up! Here's a [blog](https://developer.vonage.com/en/blog/drinking-water-reminder-app-with-node-react-and-vonage-messages-api) on how to create that app.

Reviewer 2: ```suggestion
1. **Hydration Manager**: Thirst isn't always a reliable early indicator of the body's need for water. Many people don't feel thirsty until they're already dehydrated. To make sure you stay hydrated this summer, you can [build an app using the Vonage Messages API]((https://developer.vonage.com/en/blog/drinking-water-reminder-app-with-node-react-and-vonage-messages-api) ) that sends friendly reminders to drink up!
```

> 2. **UV Index Forecaster**: After experiencing a couple of painful sunburns, I've learned that the UV index is not just a number but a crucial health indicator. It is a scale from 0 to 11+ that measures the sun's intensity which can cause skin damage and cancer when exposed without protection. Being aware of UV radiation levels can help you better plan your outdoor activities and take the right measures such as wearing sunscreen, sunglasses, and protective clothing.

Reviewer: ```suggestion
2. **UV Index Forecaster**: After experiencing a couple of painful sunburns, I've learned that the UV index is not just a number but a crucial health indicator. It is a scale from 0 to 11+ that measures the sun's intensity which can cause skin damage and cancer when exposed without protection. Being aware of UV radiation levels can help you better plan your outdoor activities and take the right measures, such as wearing sunscreen, sunglasses, and protective clothing.
```

> 2. **UV Index Forecaster**: After experiencing a couple of painful sunburns, I've learned that the UV index is not just a number but a crucial health indicator. It is a scale from 0 to 11+ that measures the sun's intensity which can cause skin damage and cancer when exposed without protection. Being aware of UV radiation levels can help you better plan your outdoor activities and take the right measures such as wearing sunscreen, sunglasses, and protective clothing.

Reviewer: ```suggestion
2. **UV Index Forecaster**: After experiencing a couple of painful sunburns, I've learned that the UV index is not just a number but a crucial health indicator. It is a scale from 0 to 11+ that measures the sun's intensity which can cause skin damage and cancer when exposed without protection. Being aware of UV radiation levels can help you better plan your outdoor activities and take the right measures, such as wearing sunscreen, sunglasses, and protective clothing.
```

TL;DR _wear sunscreen_ and build an app with the [OpenUV API](https://www.openuv.io/) that provides real-time information about UV radiation levels, encouraging users to be sun smart!

> 3. **Camping Companion**: Living in Colorado, I'm spoiled with numerous camp sites. Using the [National Park Service API](https://www.nps.gov/subjects/developer/api-documentation.htm), a camping app that provides camping recipes, safety information, and a checklist for camping gear would make any novice camper feel like a pro.

Reviewer: ```suggestion
3. **Camping Companion**: Living in Colorado, I'm spoiled with numerous campsites. Using the [National Park Service API](https://www.nps.gov/subjects/developer/api-documentation.htm), a camping app that provides camping recipes, safety information, and a checklist for camping gear would make any novice camper feel like a pro.
```

> 4. **Grill Guru**: Who doesn't love grilling in summer? An app using the [Spoonacular API](https://spoonacular.com/food-api) would not only suggest delicious grill recipes but also throw in a few summer drink ideas to quench that summer thirst–just make sure to stay hydrated with water (See 1).

Reviewer: Is it an 'app' or an 'API'?

Reviewer 2: You create that app using the API

Reviewer: I think my comment went to the wrong section. I mean this part here:  "a camping app that provides camping recipes," The explanation comes right after the link of an API [National Park Service API](https://www.nps.gov/subjects/developer/api-documentation.htm). Didn't you mean a camping API? Or there is a verb missing: e.g., ' build a campign app'  etc
```suggestion
4. **Grill Guru**: Who doesn't love grilling in summer? An app using the [Spoonacular API](https://spoonacular.com/food-api) would not only suggest delicious grill recipes but also throw in a few summer drink ideas to quench that summer thirst–just make sure to stay hydrated with water (See 1).
```

Reviewer 2: Ahh, yes, I needed a verb in there. Thank you :)

5. **Air Quality Tracker**: Some summers, I feel like I'm allergic to the air. But with an air quality tracker app using the [Breezometer API](https://breezometer.com/), users can get insights about air quality, nearby wildfires, and pollen - helping them plan their day better and receiving insights about the cleanest route to take.

> 6. **Climbing Journal**: I am not a runner. *The only running I do is from my problems*. So I was happy to find an alternative cardio workout: rock climbing! A recent [study](https://pubmed.ncbi.nlm.nih.gov/26247548/) even indicates it could help with lower back pain. This study shows that climbing can serve as a cheaper alternative to physical therapy. By using the [OpenBeta API](https://openbeta.io/), you can create an app to find climbing routes and share experiences and tips with fellow climbers.

Reviewer: As per our guide https://learn.vonage.com/contributing/: For links, try to have the link naturally included in an appropriate phrase. Rather than using “You can find more information here” or “Click here to read more”, use something like “The SMS API allows you to quickly send messages.”

> 6. **Climbing Journal**: I am not a runner. *The only running I do is from my problems*. So I was happy to find an alternative cardio workout: rock climbing! A recent [study](https://pubmed.ncbi.nlm.nih.gov/26247548/) even indicates it could help with lower back pain. This study shows that climbing can serve as a cheaper alternative to physical therapy. By using the [OpenBeta API](https://openbeta.io/), you can create an app to find climbing routes and share experiences and tips with fellow climbers.

Reviewer 2: ```suggestion
6. **Climbing Journal**: I am not a runner. *The only running I do is from my problems*. So I was happy to find an alternative cardio workout: rock climbing! [It could help with lower back pain](https://pubmed.ncbi.nlm.nih.gov/26247548/) which I definitely suffer from as someone sitting at my desk all day. This study shows that climbing can serve as a cheaper alternative to physical therapy. By using the [OpenBeta API](https://openbeta.io/), you can create an app to find climbing routes and share experiences and tips with fellow climbers.
```

7. **Personal Fitness Coach**: Need that big brain energy? Exercise has been proven to improve brain function because it stimulates the production of chemicals in the brain that helps brain cells and blood vessels grow and survive. Build an app using the [Strava API](https://developers.strava.com/) to create personalized training plans as a fantastic motivator for anyone wanting to stay active and improve their memory and thinking skills this summer.

8. **Plant Parent App**: As a proud (though horrible) plant parent, I've had my fair share of plant blunders. An app built with the [Perenual API](https://perenual.com/docs/api) would save many plants by giving advice on care, sunlight requirements, and much more.

> 9. **Geocaching Guide**: Geocaching is a high-tech global treasure hunt that involves problem-solving and exploring the outdoors. Players locate hidden containers, or "geocaches," using GPS coordinates. An app built with the [Groundspeak API](https://api.groundspeak.com/documentation#intro) would add to the excitement by allowing platers to uncover new locations, share their finds, and engage with others in the geocaching community.

Reviewer: What is 'platers'?

Reviewer 2: ```suggestion
9. **Geocaching Guide**: Geocaching is a high-tech global treasure hunt that involves problem-solving and exploring the outdoors. Players locate hidden containers, or "geocaches," using GPS coordinates. An app built with the [Groundspeak API](https://api.groundspeak.com/documentation#intro) would add to the excitement by allowing players to uncover new locations, share their finds, and engage with others in the geocaching community.
```

> 10. **3D Printing Pal**: Ever since I discovered a local makerspace, I've been fascinated by 3D printing. Using the [Thingiverse API](https://www.thingiverse.com/developers), you can build a 3D customization app where users can add text, change dimensions, or combine multiple designs, then upload their projects for other makers to try to recreate.

Reviewer 2: ```suggestion
10. **Beach Bounty Hunter**: There's nothing like finding the perfect beach spot to bask in the summer sun. With the [MagicSeaweed API](https://magicseaweed.com/developer/api), developers can craft a beach discovery app. Users can filter by preferred amenities, crowd size, or surf conditions and even share secret beach locations with the community. Create a treasure map for beach lovers everywhere!
```

## Share Your Summer Projects

> There you have it - a collection of summer-themed app ideas that combine fun in the sun with the joy of coding. Don't forget to share your summer coding adventures with us on social media! Follow and tag us on [Twitter](https://www.twitter.com/VonageDev/) to show off your work. Whether you're coding in a cozy indoor corner or taking your laptop to the beach, we want to see how you're making the most of summer with your coding skills!

Reviewer: Please add a link to our Slack Developer Community as well!

> There you have it - a collection of summer-themed app ideas that combine fun in the sun with the joy of coding. Don't forget to share your summer coding adventures with us on social media! Follow and tag us on [Twitter](https://www.twitter.com/VonageDev/) to show off your work. Whether you're coding in a cozy indoor corner or taking your laptop to the beach, we want to see how you're making the most of summer with your coding skills!

Reviewer 2: ```suggestion
There you have it - a collection of summer-themed app ideas that combine fun in the sun with the joy of coding. Don't forget to share your summer coding adventures with us on social media! Follow and tag us on [Twitter](https://www.twitter.com/VonageDev/) or join our [community Slack channel](https://developer.vonage.com/en/community/slack) to show off your work. Whether you're coding in a cozy indoor corner or taking your laptop to the beach, we want to see how you're making the most of summer with your coding skills!
```
